### PR TITLE
[llvm-opt-fuzzer] Add more passes options

### PR DESCRIPTION
**Pull request by DavidKorczynski**
The goal is to enable analysis of these passes by way of OSS-Fuzz, where they currently lack code coverage
(https://storage.googleapis.com/oss-fuzz-coverage/llvm/reports/20231225/linux/src/llvm-project/llvm/lib/Transforms/Scalar/report.html)

### CODE REVIEW

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp [hunk 1]
```diff
@@ -107,7 +108,24 @@ void llvm::handleExecNameEncodedOptimizerOpts(StringRef ExecName) {
...
+    } else if (Opt == "reassociate") {
+      Args.push_back("-passes=reassociate");
+    } else if (Opt == "lower_matrix_intrinsics") {
+      Args.push_back("-passes=lower-matrix-intrinsics");
+    } else if (Opt == "loop_reroll") {
+      Args.push_back("-passes=loop-reroll");
+    } else if (Opt == "loop_interchange") {
+      Args.push_back("-passes=loop-interchange");
+    } else if (Opt == "loop_fusion") {
+      Args.push_back("-passes=loop-fusion");
```
Comment (nikic) [inline llvm/lib/FuzzMutate/FuzzerCLI.cpp:124]: Remove these three, these are non-production passes and at least loop reroll and loop fusion are badly broken (not so sure about interchange).

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp [hunk 2]
```diff
@@ -84,9 +84,10 @@ void llvm::handleExecNameEncodedOptimizerOpts(StringRef ExecName) {
       Args.push_back("-passes=simplifycfg");
     } else if (Opt == "gvn") {
       Args.push_back("-passes=gvn");
+    } else if (Opt == "newgvn") {
+      Args.push_back("-passes=newgvn");
```
Comment (nikic) [inline llvm/lib/FuzzMutate/FuzzerCLI.cpp:88]: Probably remove this one as well. It's also a non-production pass that has enough known issues that more reports are unlikely to be useful to anyone.

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp [hunk 3]
```diff
@@ -107,7 +108,24 @@ void llvm::handleExecNameEncodedOptimizerOpts(StringRef ExecName) {
       Args.push_back("-passes=loop-reduce");
```
Comment (nikic) [inline llvm/lib/FuzzMutate/FuzzerCLI.cpp:108]: Not directly related to this PR, but I noticed `-passes=loop(rotate)` up there, which doesn't make sense. This should be `-passes=loop(loop-rotate)` or just `-passes=loop-rotate`.

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp (hunk 1 above)
Comment (DavidKorczynski) [reply llvm/lib/FuzzMutate/FuzzerCLI.cpp]: Done

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp (hunk 2 above)
Comment (DavidKorczynski) [reply llvm/lib/FuzzMutate/FuzzerCLI.cpp]: Done

File: llvm/lib/FuzzMutate/FuzzerCLI.cpp (hunk 3 above)
Comment (DavidKorczynski) [reply llvm/lib/FuzzMutate/FuzzerCLI.cpp:106]: Fixed -- set it to `-passes=loop-rotate`

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**